iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -72,29 +115,55 @@ static LogicalResult setContractConfig(FuncOp entryPoint, linalg::LinalgOp op) {
   int64_t tileX = 2;
   int64_t tileY = 256;
   int64_t tileK = 4;
+  IREE::Codegen::DispatchLoweringPassPipeline pipeline =
+      IREE::Codegen::DispatchLoweringPassPipeline::LLVMGPUMatmulSimt;
   SmallVector<int64_t, 3> workgroupSize = {2 * cudaWarpSize, 1, 1};
   bool isStaticSize =
       sizeM != ShapedType::kDynamicSize && sizeN != ShapedType::kDynamicSize;
   if (isStaticSize) {
-    // Special case for very small matrices.
-    if (sizeM * sizeN <= cudaWarpSize) {
-      tileX = sizeN;
-      tileY = sizeM;
-      workgroupSize = {sizeM, sizeN, 1};
+    bool foundTensorCoreConfig = false;
+    if (supportsTensorCore(entryPoint, op)) {
+      SmallVector<TileWorkgroupSizePair> TCtileSizeConfig;
+      getTensorCoreConfig(TCtileSizeConfig);
+      // Pick the best configuration where the original shape is aligned on the
+      // tile size.
+      for (TileWorkgroupSizePair &config : TCtileSizeConfig) {
+        if (sizeN % config.tileSize[1] == 0 &&
+            sizeM % config.tileSize[0] == 0) {
```

Comment:
No need to check `K` here? Is that handled somewhere else?

Reply:
Good point, added it.